docs/dev/README.md

```diff
@@ -0,0 +1,269 @@
+# Tekton Dashboard - Dev docs
+
+- Running tests
+- Troubleshooting
+
+This guide explains how to build, deploy and test the Tekton Dashboard. It covers the following topics:
+
+- [Before you begin](#before-you-begin)
+- [Pre-requisites](#pre-requisites)
+- [Checkout your fork](#checkout-your-fork)
+- [Build the frontend](#build-the-frontend)
+- [Build the backend](#build-the-backend)
+  - [Command line arguments](#command-line-arguments)
+- [Build YAML manifests](#build-yaml-manifests)
+- [Build and deploy with kustomize and ko](#build-and-deploy-with-kustomize-and-ko)
+  - [Build docker image for another GOARCH](#build-docker-image-for-another-goarch)
+- [Development server](#development-server)
+- [Run backend tests](#run-backend-tests)
+- [Run frontend tests](#run-backend-tests)
+  - [Unit tests](#unit-tests)
+  - [Linter](#linter)
+- [i18n](#i18n)
+- [Storybook](#storybook)
+- [Troubleshooting](#troubleshooting)
+- [Next steps](#next-steps)
+
+## Before you begin
+
+Many of the instructions here replicate what's in the [tektoncd/pipeline development guide](https://github.com/tektoncd/pipeline/blob/master/DEVELOPMENT.md), with a couple of caveats currently.
+
+- This project has not yet been tested with GKE
+
+## Pre-requisites
+
+In order to run the Tekton Dashboard, please make sure the requirements are met in [the install doc](../install.md) are met.
+
+You will also need the following tool in order to build the Dashboard locally and deploying it:
+1. [`go`](https://golang.org/doc/install): The language Tekton Dashboard is built in
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [Node.js & npm](https://nodejs.org/): For building and running the frontend locally. See `engines` in [package.json](../../package.json) for versions used. _Node.js 10.x is recommended_
+1. [`ko`](https://github.com/google/ko): For development. `ko` version v0.1 or higher is required for `dashboard` to work correctly
+1. [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/): For interacting with your kube cluster
+1. [`kustomize`](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md): For building the Dashboard manifests. You need a recent version - v3.5.4 is recommended
+See [here](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md#try-go) - `GO111MODULE=on go install sigs.k8s.io/kustomize/kustomize/v3` works correctly
+
+Note that there exists a bug in certain versions of `kubectl` whereby the `auth` field is missing from created secrets. Known good versions we've tested are __1.15.0__ and __1.16.2__.
+
+## Checkout your fork
+
+To check out this repository:
+
+1. Create your own [fork of this repo](https://help.github.com/articles/fork-a-repo/)
+1. Clone it to your machine:
+
+```bash
+mkdir -p ${GOPATH}/src/github.com/tektoncd
```

Comment:
Does it need to be on GOPATH anymore since the switch to modules? I haven't run into any issues with it outside GOPATH

Edit: I see the note below. I think we should remove references to GOPATH here to avoid confusion.

Reply:
fixed